File: 2021/D01.jl
using RollingFunctions
input = parse.(Int, readlines("D01.txt"))

function D01(input, w)
    input = rolling(sum, input, w)
    count = 0
    last = input[1]
    for line in input
        if last < line
            count += 1
        end
        last = line
    end
    count
end

println(D01(input, 1))
println(D01(input, 3))

File: 2021/D01.rb
input = File.read("D01.txt").split().map(&:to_i)

def D01(input, n)
    input.each_cons(n).map(&:sum).each_cons(2).count {|x,y| x < y }
end

puts D01(input, 1)
puts D01(input, 3)